src/StackExchange.Redis/ConnectionMultiplexer.cs

```diff
@@ -912,6 +913,8 @@ private static async Task<ConnectionMultiplexer> ConnectImplAsync(ConfigurationO
 
                     await Maintenance.ServerMaintenanceEvent.AddListenersAsync(muxer, logProxy).ForAwait();
 
+                    logProxy?.WriteLine($"Total connect time: {sw.ElapsedMilliseconds:n0} ms");
```

Comment:
Is there any value in knowing how long it took to _fail_ to connect? If so, you could move these "Total connect time" lines into the `finally`

Reply:
I debated that as well - I feel like that might be confusing to give a total time when it just blew sky high for no reason - in those cases we care more about the log of what went sky high or what was the last thing logged, rather than the time...I think.